[RFC] Ex mode: use getexline() instead of getexmodeline()

justinmk opened the pull request:
This is a departure from traditional Ex mode (Q, -e), but it should not
impact most Ex scripts.

Before this change:
* the screen test hangs.

After this change:
* Q acts like gQ.
* -e differs from -E only in its treatment of stdin.

This moves towards potentially removing getexmodeline().
(HINT: That does NOT mean "removing Ex mode", it means removing the
Vi-compatible Ex mode, which differs from Vim's "improved Ex mode" only
in some minor details (e.g. mappings are disabled).)

ref #1089 :-)~

justinmk committed:
Ex mode: use getexline() instead of getexmodeline()

This changes Ex mode (Q, -e) to work like Vim's "improved Ex mode"
(gQ, -E).  That brings some small behavior differences, but should not
impact most Ex scripts (unless, for example, they depend on mappings
being disabled--but that can be solved for -e by skipping user config).

Before this change:
* the screen test hangs.

After this change:
* Q acts like gQ.
* -e/-es differs from -E/-Es only in its treatment of stdin.

This moves towards potentially removing getexmodeline().
(HINT: That does NOT mean "removing Ex mode", it means removing the
Vi-compatible Ex mode, which differs from Vim's "improved Ex mode" only
in some minor details (e.g. mappings are disabled).)

ref #1089 :-)~

bfredl commented:
Maybe either `plain_vgetc` or `getexline` can be used in the `exmode_active` branch of do_sub so that `getexmodeline()` can be completely removed?

justinmk committed:
startup: fix -E/-Es without `-u NONE`

Before this change, -E/-Es without `-u NONE` reads stdin as Ex commands.
It should always read stdin as text (into buffer 1), like this:

    echo foo | nvim -Es +'%p'
    foo
    echo foo | nvim -Es -u NORC +'%p'
    foo
